### PR TITLE
Bugfix for `SimplexBijector` on `Matrix`

### PR DESCRIPTION
**Pull request by torfjelde**
Related: https://github.com/TuringLang/Turing.jl/issues/2190

### CODE REVIEW

**Comment by torfjelde** [comment]
@yebai @devmotion should be a quick merge

File: src/bijectors/simplex.jl [hunk 1]
```diff
@@ -136,6 +136,12 @@ function logabsdetjac(b::SimplexBijector, x::AbstractVector{T}) where {T}
 
     return -lp
 end
+
+# Needed to avoid falling back to `with_logabsdet_jacobian` for matrix inputs.
```
Comment (devmotion) [inline src/bijectors/simplex.jl:140]: Which calls `logabsdetjac` again causing a stackoverflow error (had to check the code to realize this is the underlying problem).

**Review by devmotion — APPROVED** [review]
Looks good to me (the problem was not clear to me initially but I think I understood now).

The PR made me wonder whether it would be possible to improve performance of `with_logabsdet_jacobian` for `SimplexBijector` by not performing `transform` and `logabsdetjac` separately when both are requested. Doesn't block this PR and maybe would lead to more code duplications though.

**Comment by torfjelde** [comment]
> The PR made me wonder whether it would be possible to improve performance of with_logabsdet_jacobian for SimplexBijector by not performing transform and logabsdetjac separately when both are requested. Doesn't block this PR and maybe would lead to more code duplications though.

Definitively! But yeah, probs for another PR (I'll make issue) :+1:

**Pull request merged**